### PR TITLE
Add test suite

### PR DESCRIPTION
**Pull request by FFY00**
Signed-off-by: Filipe Laíns <lains@riseup.net>

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+jobs:
+  run-tests:
+    runs-on: ubuntu-latest
+    container:
+      image: archlinux
+    steps:
+    - name: Update the system and install dependencies
+      run: pacman -Syu --noconfirm --noprogressbar --needed python-pip ninja gcc git
+
+    - name: Install ninja_syntax
```
Comment (perigoso) [inline .github/workflows/test.yml:20]: typo

Comment (perigoso) [inline .github/workflows/test.yml:20]: typo

**Review by perigoso — APPROVED** [review]

**Comment by FFY00** [comment]
I am gonna have to do a second pass at this as our current data structures are too complex to use with [ctypes](https://docs.python.org/3/library/ctypes.html). Instead of a plain shared object, I will have a native Python module that will wrap the data structures and functions with a nice Python API, which we can then use in the tests.

**Comment by FFY00** [comment]
I think this is ready, we can merge it and then start writing proper tests.

**Review by perigoso — APPROVED** [review]

**Pull request merged**